src/rdkafka_conf.c

```diff
@@ -923,6 +923,23 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
 		} },
         { _RK_TOPIC | _RK_PRODUCER, "compression.type", _RK_C_ALIAS,
           .sdef = "compression.codec" },
+	{ _RK_TOPIC|_RK_PRODUCER, "compression.level", _RK_C_INT,
+	  _RKT(compression_level),
+	  "Compression level parameter for algorithm selected by configuration "
+	  "property `compression.codec`. Higher values will result in better "
+	  "compression at the cost of more CPU usage. Usable range is "
+	  "algorithm-dependent: "
+#if WITH_ZLIB
```

Comment:
You can remove the ifdefs here, I think it is fine to provide these ranges in the config doc even if corresponding support isn't included.